gui/contextMenu.py

```diff
@@ -69,6 +69,17 @@ def getMenu(cls, selection, *fullContexts):
                 if m.display(srcContext, selection):
                     amount += 1
                     texts = m.getText(itemContext, selection)
+
+                    # Check the selected item to see if it has special reasons for not showing the menu item
+                    skip_menu_item = False
+                    for sel in selection:
+                        if hasattr(sel, "_Mode__item") and texts == "Remove Module":
+                            # Don't show remove for modes, these are special modules that cannot be removed
+                            skip_menu_item = True
+
+                    if skip_menu_item == True:
+                        continue

```

Comment:
This is not the right way to do it. :) Look in `builtinContextMenus/remove.py` or whatever it's called. That file should have a method that determines if that context menu item displays or not. You can call an `isinstance()` to determine if it shows or not (you may have to import the type from eos depending on current imports). Actually, it might be better to have the context menu context set as `fittingMode` instead of `fittingModule` - that way you can easily determine which context menus are displayed depending on the context and don't have to worry about importing type definitions. You can then bypass the entirety of the whole `isEmpty` thing and set a new context for modes.

By the way, for future reference, python has a wonderful `for...else` ability. It basically replaces things where you need a boolean flag in a for loop.